### PR TITLE
login and register pages

### LINKED ISSUE #4: Setup auth
* Setup authentication using `NextAuth`.
* Create login and register forms using shadcn ui.

Example:
Screenshot 2024-03-10 at 3 15 10 PM

### PR DESCRIPTION
**Pull request by VIGNESH-2406**
Added login and register pages with input components  of shadcn ui

### CODE REVIEW

**Review by Hrish-1 — CHANGES_REQUESTED** [review]
* You should rename all occurrences of `username` to `email` since we aren't using username for login and registration process. 
* We should be validating emails and passwords.

File: app/login/page.js [hunk 1]
```diff
@@ -1,79 +1,172 @@
...
+              {...register('password')}
+              className={`border ${errors.password ? 'border-red-500' : 'border-gray-300'} px-3 py-2 rounded-md w-full`}
+            />
+            {errors.password && (
+              <span className="text-red-500 text-sm">{errors.password.message}</span>
+            )}
+          </div>
+
+          <Button type="submit" variant="outline">
+            Submit
```
Comment (Hrish-1) [inline app/login/page.js]: You should name this button `Log in` as it clearly defines the action.

File: app/register/page.js [hunk 2]
```diff
@@ -1,84 +1,188 @@
...
+              {...register('password')}
+              className={`border ${errors.password ? 'border-red-500' : 'border-gray-300'} px-3 py-2 rounded-md w-full`}
+            />
+            {errors.password && (
+              <span className="text-red-500 text-sm">{errors.password.message}</span>
+            )}
+          </div>
+
+          <Button type="submit" variant="outline">
+            Submit
```
Comment (Hrish-1) [inline app/register/page.js]: This button should be named `Create account`.

File: app/login/page.js [hunk 3]
```diff
@@ -1,79 +1,172 @@
...
+//                         Submit
+//                     </Button>
+//                 </form>
+//             </div>
+//         </div>
+//     );
+// };
+
+// export default Login;
+
```
Comment (Hrish-1) [inline app/login/page.js]: You should remove the commented code.

File: app/register/page.js [hunk 4]
```diff
@@ -1,84 +1,188 @@
...
+//                         type="submit" variant="outline">
+//                         Submit
+//                     </Button>
+//                 </form>
+//             </div>
+//         </div>
+//     );
+// };
+
+// export default register;
```
Comment (Hrish-1) [inline app/register/page.js]: Commented code should be removed.

File: app/login/page.js [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+import { Input } from '@/components/ui/input';
+
+// Define Zod schema for form validation
+const schema = z.object({
+  email: z.string().email('Invalid email format').refine(data => {
+    // Custom validation function for email
+    const isValidEmail = /\S+@\S+\.(com|in)$/.test(data);
+    return isValidEmail || 'Invalid email format';
+  }),
+  password: z.string().min(3),
```
Comment (Hrish-1) [inline app/login/page.js]: 3 characters are way to less for a password. Minimum password length should be at least six characters or more.

Comment (VIGNESH-2406) [reply app/login/page.js]: fixed

**Pull request merged**